### PR TITLE
Updated Transformer Model Attention Mask

### PR DESCRIPTION
**Pull request by bradsmith1996**
The baseline main attention mask did not match the size of the decoder input and would therefor result in a dimension mismatch when running self.transformer_decoder because the tgt when running in inference mode is only the last sequence in the source sequence (dimension in the sequence length is 1), which does not necessarily match the max_len dimension (24 for the baseline data).

This bug was determined by running preprocessed data from the DIP dataset for which the motion_prediction task was made compatible with: https://dip.is.tuebingen.mpg.de/download.php. Training over 10 epochs with a batchsize of 24 was used to qualify that the change worked using the CMU dataset (16430 training sequences, 670 validation sequences, and 805 test sequences). The results are shown in below
loss
:

I'm currently using this repository for academic research, so any feedback on whether this is indeed the correct change (or advice on what the correct change should be would be greatly appreciated). Thank you

### CODE REVIEW

**Review by dpacgopinath — APPROVED** [review]
The proposed fix seems to be the correct change. Thanks for submitting a PR!

Could you move `tgt_mask` creation (line 146-148) inside the `if self.training` condition block since that's applicable only during training? That would prevent confusion about the dimensions of `tgt_mask` in training vs inference.

**Comment by bradsmith1996** [comment]
Thank you for reviewing the PR. The changes have been created to have the masking inside of the conditional to clear up the difference between training and inference. Thanks!
